plugins/modules/configuration_item.py

```diff
@@ -0,0 +1,395 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: configuration_item
+
+author:
+  - Manca Bizjak (@mancabizjak)
+  - Miha Dolinar (@mdolin)
+  - Tadej Borovsak (@tadeboro)
+
+short_description: Manage ServiceNow configuration items
+
+description:
+  - Create, delete or update a base ServiceNow configuration item.
+  - For more information, refer to the ServiceNow configuration management documentation at
+    U(https://docs.servicenow.com/bundle/paris-servicenow-platform/page/product/configuration-management/reference/cmdb-table-property-descriptions.html).
+
+extends_documentation_fragment:
+  - servicenow.itsm.instance
+  - servicenow.itsm.sys_id
+
+seealso:
+  - module: servicenow.itsm.configuration_item_info
+
+notes:
+  - This module manages the records of the C(cmdb_ci) ServiceNow table.
+
+options:
+  state:
+    description:
+      - State of the configuration item.
+    type: str
+    choices: [ present, absent ]
+    default: present
+  name:
+    description:
+      - The name of the configuration item.
+      - Required if the configuration item does not yet exist.
+    type: str
+  short_description:
+    description:
+      - Short description of the configuration item.
+    type: str
+  sys_class_name:
+    description:
+      - ServiceNow configuration item class.
+      - The value of this parameter should point to a ServiceNow CMDB configuration
+        item table, for instance C(cmdb_ci_server).
+      - When a new configuration item needs to be created and this parameter is
+        either not provided or points to an invalid CMDB configuration item table,
+        the resulting record will have C(sys_class_name=cmdb_ci).
+      - Note that even if you provide this parameter, the module operates directly
+        on the C(cmdb_ci) ServiceNow table. This means that you will only be able
+        to set the values for the fields of the C(cmdb_ci) table, and not for any
+        extra fields that come from the table pointed to by I(sys_class_name)).
+    type: str
```

Comment:
I know the linked issue for this PR says that we should only insert things into the `cmdb_ci`  table, but maybe we could generalize this module a bit and make the `sys_class_module` the table-selecting parameter? This way, users can create any CMDB record even with this single module.

As far as I can tell, the code does not need to change much. We need to make sure we select the right table to insert the record into. We would also probably need to fetch the exiting items in two steps (first we fetch it from the `cmdb_ci` table to get the class of the item, then fetch it from the right table to get all the fields) and make sure the classes do not diverge, but this is about it.

What do you think about that?

Reply:
I love the idea, thank you for bringing it up. I was already tempted to go in the suggested direction, because trying to explain all the limitations in the docs for the `sys_class_name` felt was almost as hard as resolving them ;)